src/main/python/hydra/lib/hydrabase.py

```diff
@@ -46,14 +46,15 @@ class AppGroup(object):
     group_name:     group name
     analyser:       Analyser class "name" e-g HAnalyser not HAnalyser()
     """
-    def __init__(self, hydra, app_name, group_name, analyser=None):
+    def __init__(self, hydra, app_name, group_name, num_app_instances, analyser=None):
```

Comment:
This does not belong here, hydra can have many applications it's ambiguous as to which one is being mentioned. 
Also applications are created later by api calls, does not make sense to have it in the constructors when no applications have been created yet.
Please remove this from here.